### PR TITLE
feat/ Implemented LIC4 & unit test

### LINKED ISSUE #6: Implement LIC4
Implement associated test case.

### PR DESCRIPTION
**Pull request by empazi**
Implemented LIC4 and its respective unit tests. 
Added exception handling to App.java to allow for LICS to throw exceptions

fixes #6

### CODE REVIEW

**Comment by Omar-AV** [comment]
Conflict currently exists so review is on hold until resolved.

**Comment by emilsjol** [comment]
When running tests on this branch locally, I get that 1 test is skipped. This doesn't seem to be the case for one other branch I tested (issue/3) or the main branch. To ensure this branch works I propose this should be fixed before merging

File: decide/src/main/java/group17/App.java
Comment (emilsjol) [inline decide/src/main/java/group17/App.java:1]: What is the reasoning for rewriting App.java in this way so that evaluateLics throws exception? Previously the lics themselves threw exceptions,

File: decide/src/main/java/group17/LicAnalyzer.java
Comment (emilsjol) [inline decide/src/main/java/group17/LicAnalyzer.java:1]: The way exceptions are thrown here doesn't provide a lot of information when the exception occurs. Compare to other lics, where there is a line included.

File: decide/src/main/java/group17/LicAnalyzer.java
Comment (emilsjol) [inline decide/src/main/java/group17/LicAnalyzer.java:156]: Potential index out of bounds if Q_PTS == NUMPOINTS and i > 0.

**Review by emilsjol — CHANGES_REQUESTED** [review]
I'm unsure about the decision regarding how to throw exceptions, but if it can be merged with no problems I'm fine with it. I think there is a potential index out of bounds exception in the nested for loop.

File: decide/src/main/java/group17/App.java
Comment (empazi) [reply decide/src/main/java/group17/App.java:1]: When I added the try/catch block I just added throws on all related methods, but it is indeed unnecessary, I removed it now.

**Review by emilsjol — APPROVED** [review]
Thanks for takes notes into consideration. Good job!

**Pull request merged**
